### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by daxpedda**
Random improvements I was doing while reviewing the code. I attempted to make descriptive commit titles.

I did increase the complexity of the `impl_traits_for` macro, but overall it made the main code cleaner I hope, I also documented it. This should be a proc-macro packed into a public crate at this point.

### CODE REVIEW

File: src/impls.rs [hunk 1]
```diff
@@ -5,139 +5,122 @@
...
 
-        impl$(<$($gen$(: $bound1$(+ $bound2)*)?),+>)? Drop for $name$(<$($gen),+>)?
+        impl$(<$($gen$(: $bound1 $(+ $bound2)*)?),+>)? Drop for $name$(<$($gen),+>)?
         {
             fn drop(&mut self) {
-                #[allow(unused_imports)]
-                use zeroize::Zeroize;
-                impl_zeroize_field_skip_pd!(self, $field1$(, $pd1)?);
-                $(impl_zeroize_field_skip_pd!(self, $field2$(, $pd2)?);)*
+                zeroize::Zeroize::zeroize(self);
```
Comment (daxpedda) [inline src/impls.rs:117]: This should do exactly the same, right?

File: src/impls.rs [hunk 2]
```diff
@@ -152,62 +135,21 @@ macro_rules! impl_serialize_and_deserialize_for {
...
-                                        core::stringify!($name)
-                                    ),
-                                )
-                            })
-                        }
-                    }
-                    deserializer.deserialize_bytes(ByteVisitor$(::<$($gen),+> (
-                        core::marker::PhantomData,
-                    ))?)
+                    Self::deserialize(<&[u8]>::deserialize(deserializer)?)
```
Comment (daxpedda) [inline src/impls.rs:149]: I'm unsure why de-serialization was so complicated before, I reduced it to this, am I missing anything?

Comment (kevinlewi) [reply src/impls.rs:149]: Hmm, I believe you added the code previously, so I am not sure why it was done that way, but the change looks good!

File: src/impls.rs (hunk 1 above)
Comment (kevinlewi) [reply src/impls.rs:117]: Indeed it does, I confirmed and also pushed a commit on top of this which tests that zeroize is run properly for each of the structs.

**Review by kevinlewi — APPROVED** [review]
This is great, thank you! I added a commit which includes tests for zeroize on the structs. Let me know if this looks ready to merge.

File: src/impls.rs (hunk 2 above)
Comment (daxpedda) [reply src/impls.rs:149]: Funny, I have no clue why I did it that way :laughing:

**Pull request merged**